src/IndexBuilder.php

```diff
@@ -11,43 +11,31 @@
 
 namespace JoliCode\Elastically;
 
-use Elastica\Exception\InvalidException;
 use Elastica\Exception\RuntimeException;
 use Elastica\Index;
 use Elastica\Reindex;
 use Elastica\Request;
 use Elastica\Response;
 use Elastica\Task;
 use Elasticsearch\Endpoints\Cluster\State;
-use Symfony\Component\Yaml\Yaml;
+use JoliCode\Elastically\Mapping\MappingProviderInterface;
 
 class IndexBuilder
 {
+    private MappingProviderInterface $mappingProvider;
     private Client $client;
-    private string $configurationDirectory;
     private IndexNameMapper $indexNameMapper;
 
-    public function __construct(Client $client, string $configurationDirectory, IndexNameMapper $indexNameMapper)
+    public function __construct(MappingProviderInterface $mappingProvider, Client $client, IndexNameMapper $indexNameMapper)
     {
+        $this->mappingProvider = $mappingProvider;
         $this->client = $client;
-        $this->configurationDirectory = $configurationDirectory;
         $this->indexNameMapper = $indexNameMapper;
     }
 
-    public function createIndex(string $indexName, ?string $fileName = null): Index
+    public function createIndex(string $indexName, array $context = []): Index
```

Comment:
This is usually a BC Break and is not done in minor version release.

Reply:
Yes, sorry about that. We had a lengthy internal discussion about bumping a major version for this and decided against it. We are going to be more careful next time :+1: